### PR TITLE
[Refactor]: Retry client

### LINKED ISSUE #32: [Refactor]: Curp client
## Description

Currently there are serveral issues in the curp client:

- Heavy code coupling, insufficient abstraction, no clear structure.
- Different logic from the Xline of the code

## Task

- [x]  unary client
- [ ]  retry clinet
- [ ] use in business client
- [ ] solate state

### PR DESCRIPTION
**Pull request by LingKa28**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
Add retry client.

* what changes does this pull request make?
  * add retry clinet.

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
No.

### CODE REVIEW

File: pkg/curp/retry.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+// Backoff config
+type backoffConfig struct {
+	// A fixed delay backoff
+	fixed *struct{}
+	// A exponential delay backoff
+	exponential *struct {
+		// Control the max delay of exponential
+		maxDelay time.Duration
+	}
+}
```
[the remark this answers is not in the document]
Comment (LingKa28) [reply pkg/curp/retry.go]: Done.

File: pkg/curp/retry.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+		return nil
+	}
+	b.count--
+	cur := b.curDelay
+	if b.config.backoff.exponential != nil {
+		b.curDelay *= 2
+		if b.curDelay > b.config.backoff.exponential.maxDelay {
+			b.curDelay = b.config.backoff.exponential.maxDelay
+		}
+	}
```
[the remark this answers is not in the document]
Comment (LingKa28) [reply pkg/curp/retry.go]: Done.

**Review by k7Xm9pQr — APPROVED** [review]
LGTM

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Pull request merged**